middleware/metric: borked

The `go vet` cleanup killed metrics (I expected so much, but didn't check). It actually crashes on startup:
~~~
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4de096]

goroutine 1 [running]:
panic(0x1b126a0, 0xc42000c180)
	/home/miek/upstream/go/src/runtime/panic.go:500 +0x1a1
sync.(*Once).Do(0x0, 0xc4205df8a0)
	/home/miek/upstream/go/src/sync/once.go:36 +0x26
github.com/miekg/coredns/middleware/metrics.setup(0xc42009c240, 0x3, 0x1cfb3f0)
	/home/miek/g/src/github.com/miekg/coredns/middleware/metrics/setup.go:33 +0x227
github.com/mholt/caddy.executeDirectives(0xc420242340, 0x1cf83f2, 0x8, 0x286b260, 0x11, 0x11, 0xc420430000, 0x1, 0x1, 0x0, ...)
	/home/miek/g/src/github.com/mholt/caddy/caddy.go:561 +0x597
github.com/mholt/caddy.startWithListenerFds(0x281d980, 0xc4201ac5c0, 0xc420242340, 0x0, 0x64, 0x0)
	/home/miek/g/src/github.com/mholt/caddy/caddy.go:459 +0x3ab
github.com/mholt/caddy.Start(0x281d980, 0xc4201ac5c0, 0x281d980, 0xc4201ac5c0, 0x0)
	/home/miek/g/src/github.com/mholt/caddy/caddy.go:427 +0xff
github.com/miekg/coredns/coremain.Run()
	/home/miek/g/src/github.com/miekg/coredns/coremain/run.go:77 +0x16f
main.main()
	/home/miek/g/src/github.com/miekg/coredns/coredns.go:6 +0x14
~~~
fix should be fairly simple though

--- a/middleware/metrics/setup.go
+++ b/middleware/metrics/setup.go
@@ -79,6 +79,6 @@ func prometheusParse(c *caddy.Controller) (Metrics, error) {
 	return met, err
 }
 
-var metricsOnce *sync.Once
+var metricsOnce sync.Once
 
 const addr = "localhost:9153"

--- a/middleware/metrics/setup_test.go
+++ b/middleware/metrics/setup_test.go
@@ -1,0 +1,28 @@
+package metrics
+
+import (
+	"testing"
+
+	"github.com/mholt/caddy"
+)
+
+func TestPrometheus(t *testing.T) {
+	tests := []struct {
+		input     string
+		shouldErr bool
+	}{
+		{`prometheus`, false},
+		{`prometheus {}`, false},   // TODO(miek): should be true
+		{`prometheus /foo`, false}, // TODO(miek): should be true
+		{`prometheus localhost:53`, false},
+	}
+	for i, test := range tests {
+		c := caddy.NewTestController("dns", test.input)
+		err := setup(c)
+		if test.shouldErr && err == nil {
+			t.Errorf("Test %v: Expected error but found nil", i)
+		} else if !test.shouldErr && err != nil {
+			t.Errorf("Test %v: Expected no error but found error: %v", i, err)
+		}
+	}
+}

